### DISCUSSION TITLE
SSR Example

### DISCUSSION THREAD

**Discussion by maxwaiyaki** [alpinejs/alpine#544] [4. General]
Is there an SSR example for Alpine.js??

**Comment by ryangjchandler** [comment]
Hi @maxwaiyaki, Alpine doesn't have any SSR support since the original design idea / philosophy was to use it with your existing server side / rendered markup. Just sprinkling bits of reactivity on top. 

Since then `x-for` has been added, but there's no solid approach for using `x-for` with existing lists.

**Reply by Angelelz** [reply]
Is this still the case in 2024?

**Comment by acrodrig** [comment]
This is what I am doing for SSR (using JSDOM), it is rather clunky though.

```typescript
Object.assign(global, { navigator: new JSDOM().window.navigator });
Object.assign(global, { MutationObserver: class { observe() {} } });

// eslint-disable-next-line @typescript-eslint/no-var-requires
const Alpine = require("alpinejs");
const xml = fs.readFileSync(__dirname + "/../test/simple.xml");
const { window } = new JSDOM(xml, { contentType: "application/xml" });

Object.assign(global, { window, document: window });

const el = window.document.documentElement;
Alpine.initializeComponent(el, {});
```

**Reply by devgauravjatt** [reply]
ReferenceError: MutationObserver is not defined

**Reply by devgauravjatt** [reply]
give a full code with html

**Comment by dashpilot** [comment]
I've just released [Alpinejs SSR](https://github.com/dashpilot/alpinejs-ssr) which is simple to use and also supports Alpine.js hydration/interactivity after server-side-rendering.
